### PR TITLE
Initial auto-generated gcloud-java-vision

### PR DESCRIPTION
**Pull request by garrettjonesgoogle**
FYI @michaelbausor @shinfan

### CODE REVIEW

File: gcloud-java-vision/src/test/java/com/google/cloud/vision/spi/v1/ImageAnnotatorTest.java [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package com.google.cloud.vision.spi.v1;
+
+import com.google.api.gax.core.PageAccessor;
```
Comment (garrettjonesgoogle) [inline gcloud-java-vision/src/test/java/com/google/cloud/vision/spi/v1/ImageAnnotatorTest.java]: @shinfan FYI there are several unused imports in the auto-generated test code

Comment (shinfan) [inline gcloud-java-vision/src/test/java/com/google/cloud/vision/spi/v1/ImageAnnotatorTest.java]: Right now all these import are statically determined so depending on the tests some of them may be unused. We could either dynamically determine imports during generation or add a post-processor to remove unused ones.

File: gcloud-java-core/pom.xml [hunk 2]
```diff
@@ -106,7 +106,7 @@
     <dependency>
       <groupId>com.google.api</groupId>
       <artifactId>gax</artifactId>
-      <version>0.0.14</version>
+      <version>0.0.15</version>
```
Comment (mziccard) [inline gcloud-java-core/pom.xml]: Build is failing as this version of gax does not exist. Did you release it?

Comment (garrettjonesgoogle) [inline gcloud-java-core/pom.xml]: I had assumed it was released, but I was wrong. It is released now.

File: gcloud-java-vision/src/test/java/com/google/cloud/vision/spi/v1/ImageAnnotatorTest.java (hunk 1 above)
Comment (mziccard) [inline gcloud-java-vision/src/test/java/com/google/cloud/vision/spi/v1/ImageAnnotatorTest.java]: This should be done, there's a big amount of used imports in all generated code.

File: gcloud-java-vision/src/main/java/com/google/cloud/vision/spi/v1/ImageAnnotatorApi.java [hunk 3]
```diff
@@ -0,0 +1,222 @@
...
+   *     .addAllRequests(requests)
+   *     .build();
+   *   BatchAnnotateImagesResponse response = imageAnnotatorApi.batchAnnotateImages(request);
+   * }
+   * </code></pre>
+   *
+   * @param request The request object containing all of the parameters for the API call.
+   * @throws com.google.api.gax.grpc.ApiException if the remote call fails
+   */
+  private final BatchAnnotateImagesResponse batchAnnotateImages(
```
Comment (mziccard) [inline gcloud-java-vision/src/main/java/com/google/cloud/vision/spi/v1/ImageAnnotatorApi.java]: Is this meant to be private?

Comment (garrettjonesgoogle) [inline gcloud-java-vision/src/main/java/com/google/cloud/vision/spi/v1/ImageAnnotatorApi.java]: Yes, our code generator automatically hides the request-based methods when the request has only one field, so the flattened version is more direct. Will this cause a problem? We can re-evaluate.

Comment (mziccard) [inline gcloud-java-vision/src/main/java/com/google/cloud/vision/spi/v1/ImageAnnotatorApi.java]: No this is fine. Just wanted to be sure.

File: gcloud-java-vision/src/test/java/com/google/cloud/vision/spi/v1/ImageAnnotatorTest.java (hunk 1 above)
Comment (garrettjonesgoogle) [inline gcloud-java-vision/src/test/java/com/google/cloud/vision/spi/v1/ImageAnnotatorTest.java]: We will try to figure out a way to squash them all, but probably after this PR.

**Comment by garrettjonesgoogle** [comment]
This has been rebased, PTAL

**Pull request merged**